### PR TITLE
fix: Parse `GenerateJSONOutput.py` input args properly

### LINKED ISSUE #83: `GenerateJSONOutput.py` fails to parse inputs properly
I'm also getting this now:

```console
[ericballabene]$ GenerateJSONOutput.py -i combined_hypotest_Nominal.root -f hypo_C1C1_%f_%f
```
```pytb
>>> ... Setting:       addCoordinates                     {"m0":"x","m12":"y"}
>>> ... Setting:                  cut                                        1
>>> ... Setting:               format                          hypo_C1C1_%f_%f
Traceback (most recent call last):
  File "/users2/ballaben/histfitter_v1.0.0/scripts/GenerateJSONOutput.py", line 59, in <module>
    setting, eval("args.%s" % setting)
TypeError: unsupported format string passed to list.__format__
```

_Originally posted by @ericballabene in https://github.com/histfitter/histfitter/issues/81#issuecomment-892027074_

### PR DESCRIPTION
**Pull request by matthewfeickert**
Resolves #83 

**Suggested squash and merge commit message**:
```
* Properly parse argparse arguments with `vars` and `getattr`
* Simplify imports and environmental manipulation and move to top of file
```

### CODE REVIEW

**Comment by ericballabene** [comment]
these commits solve the issue to me

File: scripts/GenerateJSONOutput.py [hunk 1]
```diff
@@ -50,25 +57,12 @@
...
-for setting in dir(args):
-    if not setting[0] == "_":
-        print(
-            ">>> ... Setting: {: >20} {: >40}".format(
-                setting, eval("args.%s" % setting)
-            )
-        )
+for arg in vars(args):
+    user_input = getattr(args, arg)
+    print(f">>> ... Setting: {arg: >20} {str(user_input): >40}")
```
Comment (matthewfeickert) [inline scripts/GenerateJSONOutput.py:63]: @histfitter/developers When the argument is a list is the desired behavior like how I have it now which would give (I think this was the previous format)

```console
$ python scripts/GenerateJSONOutput.py -i file_A.root file_B.root -f hypo_C1C1_%f_%f
>>> ... Setting:           inputFiles           ['file_A.root', 'file_B.root']
>>> ... Setting:               format                          hypo_C1C1_%f_%f
>>> ... Setting:       interpretation                                   m0:m12
>>> ... Setting:       addCoordinates                     {"m0":"x","m12":"y"}
>>> ... Setting:                  cut                                        1
>>> ... Setting:            noAddTabs                                    False
```

or is the alternative of having them one per line preferred?

```console
$ python scripts/GenerateJSONOutput.py -i file_A.root file_B.root -f hypo_C1C1_%f_%f
>>> ... Setting:           inputFiles                              file_A.root
>>> ... Setting:           inputFiles                              file_B.root
>>> ... Setting:               format                          hypo_C1C1_%f_%f
>>> ... Setting:       interpretation                                   m0:m12
>>> ... Setting:       addCoordinates                     {"m0":"x","m12":"y"}
>>> ... Setting:                  cut                                        1
>>> ... Setting:            noAddTabs                                        0
```

File: scripts/GenerateJSONOutput.py [hunk 2]
```diff
@@ -50,25 +57,12 @@
 
 args = parser.parse_args()
 
-
 # Print out the settings
-for setting in dir(args):
-    if not setting[0] == "_":
```
Comment (matthewfeickert) [inline scripts/GenerateJSONOutput.py:56]: Note sure why this check was ever needed, but happy to have it back in there if really needed

**Review by kratsg — APPROVED** [review]

**Comment by kratsg** [comment]
I know @lawrenceleejr wrote a lot of this file and probably might have some suggestions/opinions.

**Comment by matthewfeickert** [comment]
Ping @lawrenceleejr. Though I appreciate that you might not have time to look at this at the moment.

**Comment by lawrenceleejr** [comment]
Hey all -- sorry for being slow to get to this. Thanks for taking care of this -- this is definitely the right thing to do on both of these changes. Totally approve on my side!
-L

**Comment by matthewfeickert** [comment]
Thanks very much for the feedback, @lawrenceleejr! :) 

@kratsg if you and the other @histfitter/developers are happy with this then I think it can get merged whenever.

**Pull request merged**